test/ops/align.jl

```diff
@@ -1,11 +1,19 @@
 @testset "right" begin
     _test_binary_op(right, (x, y) -> y)
-    @test right(n1, n1) === n1
+    for alignment in (LEFT, UNION, INTERSECT)
```

Comment:
Should you also be adding tests to check that arguments get converted into a `Node` if they're not already? i.e. I'm assuming that the additional `_ensure_node` calls in the definition of `left` are more of a bug fix than a new feature.

Same for `right` and other alignment operations

Reply:
> I'm assuming that the additional _ensure_node calls in the definition of left are more of a bug fix 

Actually the code before was OK, because all we did was pass `x` and `y` into `apply`, which did the conversion. Now we use `_is_empty` on them first, which requires them to definitely be nodes.

> Should you also be adding tests to check that arguments get converted into a Node if they're not already? 

Yep. It is actually already tested here, because we call `_test_binary_op` above:

https://github.com/tpgillam/TimeDag.jl/blob/ba8c0e54598a050d6db378a4833261b0e9eee651/test/common.jl#L219-L226